### PR TITLE
SISRP-46907 - Migrate CalCentral to Student API V2

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira-secure.berkeley.edu/browse/SISRP-46907

This update involves migrating from Student API Affiliations V1 to the SIS Person API V1, because the Student API V2 doesn't support returning data for non-students. I did this within HubEdos::UserAttributes, because it's the model that provides user attributes from Campus Solutions for all users. It's combined with LDAP attributes in User::AggregatedAttributes, because we need Staff / Faculty attributes from LDAP (Campus Solutions is the student source of record, not faculty/staff).

After retrofitting those models to source user attributes from the SIS Person API + Student Attributes (if applicable), I continued to update other code that depended on Student API V1 proxies so that they would use the Student API V2 proxies. Most of that work is completed, with the exception of the `HubEdos::StudentApi::V1::WorkExperience` proxy. I'll complete that when I return.

Everything I've done is theoretically working well in Rspec test world, but I have to actual integration testing to continue working out any issues that remain.

### CODE REVIEW

File: app/models/hub_edos/student_api/v1/contacts.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module HubEdos
   module StudentApi
     module V1
+      # TODO: Remove this deprecated proxy
```
Comment (redconfetti) [inline app/models/hub_edos/student_api/v1/contacts.rb]: I will be removing these proxies once testing and bugs are resolved.

File: app/models/hub_edos/person_api/v1/sis_person.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+module HubEdos
+  module PersonApi
+    module V1
+      class SisPerson < HubEdos::Proxy
```
Comment (developish) [inline app/models/hub_edos/person_api/v1/sis_person.rb]: When inheriting from a class higher in the module chain, I like to be specify the starting point by notating it like this: `class SisPerson < ::HubEdos::Proxy`. Not necessary but it's slightly more specific

File: app/models/my_registrations/statuses.rb [hunk 3]
```diff
@@ -41,94 +30,19 @@ def get_registration_statuses
...
-        # All term registration statuses are hidden the day after the term ends.
-        pastEndOfInstruction: current_date > term[:end],
-        # Financial Aid disbursement is used in CNP notification.  This is defined as 9 days before the start of instruction.
-        pastFinancialDisbursement: current_date >= (term[:classesStart] - 9),
-      })
-    end
-
     # Match registration terms with Berkeley::Terms-defined terms.
-    def match_terms(registrations, terms)
+    def match_terms(registrations = [], terms)
```
Comment (developish) [inline app/models/my_registrations/statuses.rb]: With ordered arguments like this, using the default argument for registrations is impossible unless you also specific a default argument for terms

**Comment by developish** [comment]
I don't see any showstoppers here

**Pull request merged**